admin/app/components/smart-random-simulator/smart-random-params.js

```diff
@@ -0,0 +1,89 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import Joi from 'joi';
+
+export default class SmartRandomParams extends Component {
+  @tracked errors = {};
+
+  acceptedLocales = [
+    { value: 'en', label: 'en' },
+    { value: 'fr', label: 'fr' },
+    { value: 'fr-fr', label: 'fr-fr' },
+    { value: 'nl', label: 'nl' },
+  ];
+
+  validationSchema = {
+    skills: Joi.array()
+      .items({
+        id: Joi.string().required(),
+        difficulty: Joi.number().integer().min(1).max(8).required(),
+        name: Joi.string().required(),
+      })
+      .min(1)
+      .required(),
+    challenges: Joi.array()
+      .items({
+        id: Joi.string().required(),
+        timer: Joi.number(),
+        skill: {
+          id: Joi.string().required(),
+          name: Joi.string().required(),
+        },
+        locales: Joi.array()
+          .items(Joi.string().valid(...this.acceptedLocales.map((locale) => locale.value)))
+          .required(),
+      })
+      .min(1)
+      .required(),
+    knowledgeElements: Joi.array()
+      .items({
+        source: Joi.string().valid('direct', 'inferred').required(),
+        status: Joi.string().valid('validated', 'invalidated', 'reset').required(),
+        'answer-id': Joi.string().required(),
+        skillId: Joi.string().required(),
```

Comment:
```suggestion
        'skill-id': Joi.string().required(),
```
Sinon Joi me demande `skillId`.
Alors que quand je fais "Démarrer la simulation", on me demande `skill-id`.

Reply:
ah zut